test/e2e/csv_e2e_test.go

```diff
@@ -530,11 +530,16 @@ func TestCreateCSVRequirementsMetCRD(t *testing.T) {
 		},
 	}
 
-	// Create dependency first (CRD)
+	// Create CSV first, knowing it will fail
+	cleanupCSV, err := createCSV(t, c, crc, csv, testNamespace, true)
+	require.NoError(t, err)
+	defer cleanupCSV()
+
+	fetchedCSV, err := fetchCSV(t, crc, csv.Name, csvPendingChecker)
```

Comment:
You should attempt to fetch with _any_ condition since you can't guarantee it will still be pending at this point.